x-pack/platform/plugins/shared/lens/public/react_embeddable/initializers/initialize_alert_rules.test.tsx

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { last } from 'lodash';
+import React from 'react';
+import type { ActionTypeRegistryContract, RuleTypeRegistryContract } from '@kbn/alerts-ui-shared';
+import { render, screen } from '@testing-library/react';
+import type { ReactNode } from 'react';
+import { BehaviorSubject } from 'rxjs';
+import { createUnifiedSearchApi, createEsqlVariablesApi, makeEmbeddableServices } from '../mocks';
+import { initializeAlertRules } from './initialize_alert_rules';
+import { ESQLVariableType } from '@kbn/esql-types';
+
+const mockRender = render;
+const mockRuleFormFlyout = jest.fn((props) => <div data-test-subj={props['data-test-subj']} />);
+jest.mock('@kbn/react-kibana-mount', () => ({
+  toMountPoint: jest.fn((node: ReactNode) => mockRender(node)),
+}));
+
+jest.mock('@kbn/response-ops-rule-form/flyout', () => ({
+  RuleFormFlyout: (...args: Parameters<typeof mockRuleFormFlyout>) => mockRuleFormFlyout(...args),
+}));
+const ruleTypeRegistry: jest.Mocked<RuleTypeRegistryContract> = {
+  has: jest.fn(),
+  register: jest.fn(),
+  get: jest.fn(),
+  list: jest.fn(),
+};
+const actionTypeRegistry: jest.Mocked<ActionTypeRegistryContract> = {
+  has: jest.fn(),
+  register: jest.fn(),
+  get: jest.fn(),
+  list: jest.fn(),
+};
+const parentApiMock = {
+  ...createUnifiedSearchApi(),
+  ...createEsqlVariablesApi(),
+};
+
+const startDependenciesMock = makeEmbeddableServices(new BehaviorSubject<string>(''), undefined, {
+  visOverrides: { id: 'lnsXY' },
+  dataOverrides: { id: 'formBased' },
+});
+
+const uuid = '000-000-000';
+
+const getLastCalledInitialValues = () => last(mockRuleFormFlyout.mock.calls)![0].initialValues;
+
+describe('Alert rules API', () => {
+  const { api } = initializeAlertRules(uuid, parentApiMock, startDependenciesMock);
+
+  describe('createAlertRule', () => {
+    it('should pass initial values to the rule form and open it', async () => {
+      api.createAlertRule(
+        {
+          params: {
+            searchType: 'esqlQuery',
+            esqlQuery: {
+              esql: "FROM were_no_strangers_to_love | STATS `you know the rules and so do i` = COUNT(*) BY `a full commitment's what i'm thinking of`, `you wouldn't get this from any other guy`",
```

Comment:
Just a small thing - while the lyrics are a fun touch, for the sake of readability and maintainability, could we perhaps replace them with some placeholder text or realistic sample data? It would make the tests a bit clearer too. Thanx!

Reply:
One of these days I will get a rickroll through code review, mark my words

Reply:
Thanx for that Zac, fixed here https://github.com/elastic/kibana/pull/217719/commits/994e296c71d3ac34de2d331e06de7da1cf2bfc99

I didnt want to cause a heart break, if it wasn't a public repo I would be very happy to have it